runway/hooks/staticsite/auth_at_edge/client_updater.py

```diff
@@ -0,0 +1,117 @@
+"""User Pool Client Updater.
+
+Responsible for updating the User Pool Client with the generated
+distribution url + callback url paths.
+"""
+
+import logging
+# pylint: disable=unused-import
+from typing import Any, Dict, Optional
+
+import boto3
+
+from ....cfngin.lookups.handlers.output import OutputLookup
+from ....cfngin.lookups.handlers.rxref import RxrefLookup
+# pylint: disable=unused-import
+from ....cfngin.providers.base import BaseProvider
+from ....cfngin.session_cache import get_session
+
+LOGGER = logging.getLogger(__file__)
+
+
+def update(context,  # type: context.Context
+           provider,  # type: BaseProvider
+           **kwargs  # type: Optional[Dict[str, Any]]
+           ):
+    # type: (...) -> bool
+    """Update the callback urls for the User Pool Client.
+
+    Required to match the redirect_uri being sent which contains
+    our distribution and alternate domain names.
+
+    Args:
+        context (:class:`runway.cfngin.context.Context`): The context
+            instance.
+        provider (:class:`runway.cfngin.providers.base.BaseProvider`):
+            The provider instance
+
+    Keyword Args:
+        alternate_domains (List[str]): A list of any alternate domains
+            that need to be listed with the primary distribution domain
+        redirect_path_sign_in (str): The redirect path after sign in
+        redirect_path_sign_out (str): The redirect path after sign out
+        oauth_scopes (List[str]): A list of all available validation
+            scopes for oauth
+    """
+    session = get_session(provider.region)
+    # @TODO: Is this check necessary?
+    if session:
+        cognito_client = session.client('cognito-idp')
+    else:
+        cognito_client = boto3.client('cognito-idp')
+
+    user_pool_client = RxrefLookup.handle(
```

Comment:
this can be removed since lookups are natively resolved in hook args (#143) so you can use `client_id: ${rxref StackName::OutputName}` in your config and use `kwargs['client_id']` directly